cmds/core/readlink/readlink.go

```
@@ -49,27 +49,38 @@ func readLink(file string) error {
 		path, err = filepath.EvalSymlinks(file)
 	}
 
-	if *nonewline {
+	delimiter := "\n"
+	if *noNewLine {
 		delimiter = ""
 	}
 
-	fmt.Printf("%s%s", path, delimiter)
+	fmt.Fprintf(stdout, "%s%s", path, delimiter)
 	return err
 }
 
-func main() {
-	flag.Parse()
-
-	var exitStatus int
+func run(stdout io.Writer, stderr io.Writer, args []string) int {
```

This should return an error. Otherwise this looks pretty good.